src/Eshopworld.Core/ObjectReferenceEqualityComparer.cs

```diff
@@ -11,7 +11,7 @@
     public class ObjectReferenceEqualityComparer<T> : EqualityComparer<T>
         where T : class
     {
-        private static IEqualityComparer<T> _defaultComparer;
+        private static IEqualityComparer<T>? _defaultComparer;
```

Comment:
do we want to pre-instantiate this to avoid "?"...?

Reply:
I've simplified this. I don't think the previous optimisation made any difference